compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertRegionToWorkgroups.cpp

```diff
@@ -204,15 +201,11 @@ rewriteFlowDispatchRegionToFlowDispatchWorkgroups(
   rewriter.create<IREE::Flow::ReturnOp>(loc);
   rewriter.eraseOp(terminator);
 
-  // Create workload region.
-  if (workloadBuilder.has_value()) {
-    Region &workgroupCountRegion = workgroupsOp.getWorkgroupCount();
-    Block *body = rewriter.createBlock(&workgroupCountRegion);
-    SmallVector<BlockArgument> workloadArgs;
-    for (Value v : workload)
-      workloadArgs.push_back(body->addArgument(v.getType(), loc));
-    rewriter.setInsertionPointToStart(body);
-    workloadBuilder->regionBuilder(rewriter, loc, workloadArgs);
+  // Clone DispatchRegion's workload_count region to DispatchWorkgroupOp's
+  if (!regionOp.getWorkgroupCount().empty()) {
+    BlockAndValueMapping bvm;
+    regionOp.getWorkgroupCount().cloneInto(&workgroupsOp.getWorkgroupCount(),
```

Comment:
Since you are replacing the `flow.dispatch.region` op you should be able to move the region into the new op.

Reply:
I am not sure how can I do that.

Reply:
https://github.com/llvm/llvm-project/blob/b4d711dc968c2e9df60581de6fff89716c583d05/mlir/include/mlir/IR/PatternMatch.h#L404